src/youtube/BaseVideo/BaseVideoParser.ts

```diff
@@ -37,13 +37,11 @@ export class BaseVideoParser {
 		target.tags =
 			videoInfo.superTitleLink?.runs
 				?.map((r: YoutubeRawData) => r.text.trim())
-				.filter((t: string) => t) || [];
+				.filter((t: string) => t) || [];	
 		target.description = videoInfo.videoDetails.shortDescription || "";
 
 		// related videos
-		const secondaryContents = data.response.contents.twoColumnWatchNextResults.secondaryResults?.secondaryResults.results.find(
-			(s: YoutubeRawData) => s.itemSectionRenderer
-		).itemSectionRenderer.contents;
+		const secondaryContents = data.response.contents.twoColumnWatchNextResults.secondaryResults?.secondaryResults.results
```

Comment:
```suggestion
		const secondaryContents = data.response.contents.twoColumnWatchNextResults.secondaryResults?.secondaryResults.results.find(
			(s: YoutubeRawData) => s.itemSectionRenderer
		)?.itemSectionRenderer?.contents ?? data.response.contents.twoColumnWatchNextResults.secondaryResults?.secondaryResults.results;
```

It seemed necessary to keep what was originally obtainable without change.

Reply:
and if I put it inside an if, to check if that element exists, if not, do not do the find